vignettes/example_analysis_coxreg.Rmd

```diff
@@ -0,0 +1,380 @@
+  if (!is.null(cache_env[[cov]])) {
+    ## If model already exists, retrieve it from cache_env
+    model <- cache_env[[cov]]
+  } else {
+    ## Build model formula
+    model_form <- paste0("survival::Surv(AVAL, EVENT) ~ ARM")
+    if (length(cov) > 0) {
+      model_form <- paste(c(model_form, cov), collapse = " * ")
+    } else {
+      cov <- "ARM"
+    }
+    ## Calculate Cox regression model
+    model <- survival::coxph(
+      formula = stats::as.formula(model_form),
+      data = df,
+      ties = "exact"
+    )
+    ## Store model in the caching environment
+    cache_env[[cov]] <- model
+  }
+  model
+}
+```
+
+## Creating the Analysis Function: `a_cox_summary`
+
+With our data prepared and helper function created, we can proceed to construct our analysis function `a_cox_summary`, which will be used to populate all of the rows in our table. In order to be used to generate both data rows (for interaction effects) and content rows (for main effects), we must create a function that can be used as both `afun` in `analyze` and `cfun` in `summarize_row_groups`. Therefore, our function must accept either `x` or `df` as its first parameter and also accept the `labelstr` parameter.
+
+The arguments of our analysis function will be as follows:
+
+-   `df` - a `data.frame` of the full dataset required to fit the Cox regression model.
+-   `labelstr` - the `string` label for the variable being analyzed in the current row/column split context.
+-   `.spl_context` - a `data.frame` containing the `value` column which is used by this analysis function to determine the name of the variable/covariate in the current split. For more details on the information stored by `.spl_context` see `?analyze`.
+-   `stat` and `format` - `string`s that indicate which statistic column we are currently in and what format should be applied to print the statistic.
+-   `cache_env` - an `environment` object that can be used to store cached models so that we can prevent repeatedly fitting the same model. Instead, each model will be generated once per covariate and then reused.
+-   `cov_main` - a `logical` value indicating whether or not the current row is summarizing covariate main effects.
+
+The analysis function works within a given row/column split context by using the current covariate (`cov`) and the `cached_model` function to obtain the desired Cox regression model. From this model, the `h_coxreg_extract_interaction` function is able to extract information/statistics relevant to the analysis and store it in a `data.frame`. The rows in this `data.frame` that are of interest in the current row/column split context are then extracted and the statistic to be printed in the current column is retrieved from these rows. Finally, the formatted rows containing the value of this statistic are returned with appropriate labelling. For more detail see the commented function code below, where the purpose of each line within `a_cox_summary` is described.
```

Comment:
We don't return rows here, we return whats called a "VerticalRowsSection"  object (thats what in_rows acts as a constructor for), so please change the description of what is returned. 

VerticalRowsSection objects are basically stacks of _cells_, rather than full rows, they need to be placed side by side and then sliced horizontally to construct rows, but users defining analysis functions don't really need to know that.